cpp/modmesh/inout/gmsh.cpp

```diff
@@ -94,6 +94,159 @@ void Gmsh::build_interior(const std::shared_ptr<StaticMesh> & blk)
     blk->build_boundary();
     blk->build_ghost();
 }
+
+// Check the finite state machine state transition is valid or not to check msh file format is correct
+bool Gmsh::is_valid_transition(const std::string s)
```

Comment:
Could you please help change the argument to be a const reference?  It's a simple change but needs to be applied in both the declaration (in header file) and the definition (here).

To be specific, it is to change

```cpp
bool Gmsh::is_valid_transition(const std::string s)
```

with

```cpp
bool Gmsh::is_valid_transition(const std::string & s)
```

That is, change the argument of the function from `const std::string s` to `const std::string & s`.

The change will speed up the code by saving unnecessary string copying.